### ISSUE TITLE
pnpm global update not working

### ISSUE THREAD

**Issue by MohamadKh75** [pnpm/pnpm#3779]
### pnpm version:

6.15.0

### Code to reproduce the issue:

`pnpm update --interactive --latest --global`

### Expected behavior:

To upgrade pnpm itself to the newer version (in this case, 6.15.1)

### Actual behavior:

it shows the progress and "Already up-to-date" message.

### Additional information:

- `node -v` prints: v14.17.6
- Windows, macOS, or Linux?: Linux ⬇️

```
System:
  Host: ZenBook Kernel: 5.4.0-84-generic x86_64 bits: 64
  Desktop: KDE Plasma 5.18.5 Distro: Ubuntu 20.04.3 LTS (Focal Fossa)
```

**Comment by silver886** [comment]
Same here.

### pnpm version:

6.24.2

### Code to reproduce the issue:

pnpm up -igL

### Expected behavior:

Update selected packages to the latest version.

### Actual behavior:

Does nothing.

```
√ Choose which packages to update (Press <space> to select, <a> to toggle all, <i> to invert selection) · aws-cdk, pnpm
Already up-to-date
Progress: resolved 370, reused 369, downloaded 0, added 0, done
```

Debug information comes with `--reporter ndjson`.
[pnpm-debug.log](https://github.com/pnpm/pnpm/files/7778254/pnpm-debug.log)

### Additional information:

 - `node -v` prints: `v14.18.1`
 - Windows, macOS, or Linux?: Windows (`Microsoft Windows [Version 10.0.19044.1415]`)

**Comment by tjx666** [comment]
Same here.

<img width="798" alt="image" src="https://user-images.githubusercontent.com/41773861/154795524-1b1bd8ae-ead8-4b7f-91a2-64bde2661520.png">

**Comment by zkochan** [comment]
Try with pnpm v7

**Comment by tjx666** [comment]
> Try with pnpm v7

Still not woring in latest version 7.0.0-rc.9

<img width="810" alt="image" src="https://user-images.githubusercontent.com/41773861/165996328-27dd8bfc-2931-496f-af21-d7b743c99b66.png">


Latest vsce version is 2.7.0

**Comment by zkochan** [comment]
Yes, I could reproduce it.

This works though:

```
pnpm i --global  vsce
```

It will install the latest `vsce`.

**Comment by vemoo** [comment]
I think this is the same as #3720
I wrote the results of some investigations there: https://github.com/pnpm/pnpm/issues/3720#issuecomment-978125390, but I don't know if they are still relevant on pnpm v7.

**Comment by b0o** [comment]
I'm doing this as a workaround:

```
$ echo virtual-store-dir=.pnpm > "$PNPM_HOME/global/5/.npmrc"
$ pnpm update --dir "$PNPM_HOME/global/5" --interactive --latest
```

Might be dangerous but it seems to work.

**Comment by exsesx** [comment]
It worked for me, too, thanks @b0o 

But I wonder if there will be a fix for this problem or a best practice.

**Comment by tjx666** [comment]
@zkochan any update on this? I think this is an serious issue.

**Comment by aydgn** [comment]
I use @b0o's [way](https://github.com/pnpm/pnpm/issues/3779#issuecomment-1207514152) but it should be fixed ASAP.

**Comment by MohamadKh75** [comment]
I also noticed that when I update pnpm using brew, I lost all my global packages
Seems related to this

cc @zkochan (I really appreciate your hard work on pnpm, tnx mate ❤️ )

**Comment by ekwoka** [comment]
Yeah, doesn't work. Outdated sees it's outdated, but update just says it's all up to date, even with latest flag

**Comment by ekwoka** [comment]
Well, you can use `npm-check-updates` with the pnpm flag

```
var=$(pnpm dlx npm-check-updates -gp pnpm | grep -i "pnpm -g add")
eval "$var"
```

**Comment by lawrencejob** [comment]
I'm so glad it's not just me. I was pouring over the documentation trying to figure out what I wasn't understanding.

**Comment by kytta** [comment]
This is the same for me. `pnpm up -giL` as well as `pnpm up -gL` do so as if they're doing work: They seem to resolve the deps, and they even download something, but this never results in an upgrade. `pnpm up -g` works, but not for major updates; for this, I have to kinda re-install the package (as described in https://github.com/pnpm/pnpm/issues/3779#issuecomment-1114092222).

**Comment by ekwoka** [comment]
@kytta 

`npm-check-updates` works with pnpm and global packages. Even lets you preview and interactively select the updates.

**Comment by iamchathu** [comment]
This is seems same with version 8. What is the fix for this?

**Comment by vemoo** [comment]
This has been fixed on [v8.6.3](https://github.com/pnpm/pnpm/releases/tag/v8.6.3), I'm not sure it has been fixed on v7

**Comment by zkochan** [comment]
[7.33.2](https://github.com/pnpm/pnpm/releases/tag/v7.33.2) has the fix too
